server/src/test/java/org/elasticsearch/cluster/metadata/MetadataIndexStateServiceBatchingTests.java

```diff
@@ -71,7 +72,7 @@ public void testBatchOpenIndices() throws Exception {
         final var future2 = client().admin().indices().prepareOpen("test-2", "test-3").execute();
 
         // check the queue for the open-indices tasks
-        assertThat(masterService.pendingTasks(), hasSize(3)); // two plus the blocking task itself
+        assertThat(findPendingTasks(masterService, "open-indices"), hasSize(2));
```

Comment:
Under heavy CPU load "reconcile-desired-task" could have been delayed and sometimes this would record 4 tasks.